uaclient/util.py

```diff
@@ -752,3 +752,29 @@ def handle_message_operations(
             if not functor(**args):
                 return False
     return True
+
+
+def parse_rfc3339_date(dt_str: str) -> datetime.datetime:
```

Comment:
I think I am missing something here, but why do we need to support this RFC here ? I understand that this would be a generic approach, where the contracts team could send different date formats to us and we would be able to parse it. but maybe we don't need it at the moment.

However, if this was already discussed with the contracts team, no problem

Reply:
@lucasmoura 
This isn't that generic - it is just one specific date format (and honestly probably doesn't cover the whole spec). We need to support this date format because it is the date format that the golang time package outputs when serializing to json (and I think it is a de facto standard for date strings in json), and it would be unnecessarily complex to override that.

I just noticed we parse a date we got from contract-server when we look at effectiveTo at `config.py:532`, but that is only working because the server has been consistently not including subseconds and consistently been using UTC (`Z`). If either of those ever change, then that line will break. Perhaps it's not likely that will change for production, but if we ever set up a test contract server, it may default to including a local timezone, for example. Or if we start creating arbitrary contracts, we may (purposefully or not) end up with a date including subseconds. Generally though, I think its better for our date parsing to be just a bit more robust and handle these cases rather than crash.

Also, while looking for other places we parse dates just now, I noticed we have a `DatetimeAwareJSONDecoder` class that is currently not successfully parsing dates because it doesn't cover the timezone or subseconds.

So I think we should actually use this new `parse_rfc3339_date` function in `DatetimeAwareJSONDecoder`, and then we won't need to parse dates anywhere else.